crates/holochain/src/core/workflow/app_validation_workflow.rs

```diff
@@ -632,12 +633,33 @@ fn create_link_zomes_to_invoke(
 }
 
 /// Get the zomes to invoke for an [`Op::StoreRecord`].
-fn store_record_zomes_to_invoke(
+async fn store_record_zomes_to_invoke(
     action: &Action,
     ribosome: &impl RibosomeT,
+    cascade: &(impl Cascade + Send + Sync),
 ) -> AppValidationOutcome<ZomesToInvoke> {
+    // For deletes there is no entry type to check, so we get the previous action to see if that
+    // was a create or a delete for an app entry type.
+    let action = match action {
```

Comment:
Was there no check of the create action before?

Reply:
Create and update are checked but a delete never carries information about the entry type, so we need to look backwards from the delete to a create or update that does have entry type information. So this is a little pre-step that then ends up using the same code that was already here for creates and updates

Reply:
Oh yes, it would fall through the match block below and trigger app validation in all zomes. Which is both wasteful and confusing for implementations who are being passed types they don't know anything about